website/src/pages/dictionary/index.js

```diff
@@ -173,8 +224,16 @@ function DataDictionary() {
     );
 
     setFilters({ tiers: [...validDataTiers], attributes: [...validDataAttributes] });
+    setMenuContents(generateMenuContents(schemas));
   }, [dictionary]);
 
+  const filteredSchemas = React.useMemo(() => searchSchemas(), [searchParams]);
+
+  useEffect(() => {
+    setSchemas(filteredSchemas);
```

Comment:
why not render `filteredSchemas` directly rather than setting it in state? Assuming `filteredSchemas` is a transformed version of `schemas`, there's no need to set `filteredSchemas` as the current schema.